celery/platforms.py

```diff
@@ -781,8 +782,54 @@ def check_privileges(accept_content):
         if not all(hasattr(os, attr)
                    for attr in ['getuid', 'getgid', 'geteuid', 'getegid']):
             raise SecurityError('suspicious platform, contact support')
-
-    if not uid or not gid or not euid or not egid:
+    
+    # Get the group database entry for the current user's group and effective group id using grp.getgrgid() method
+    # We must handle the error in the event of gid or egid not being found.
+    try:
+        gid_entry = grp.getgrgid(gid)
+        egid_entry = grp.getgrgid(egid)
+    except KeyError:
+        print('An entry for the specified gid or egid was not found')
+        if ('pickle' in accept_content or
+                'application/x-python-serialize' in accept_content):
```

Comment:
We're testing this twice. Both here and in line 817-818.
We should extract the test to a variable.

Reply:
@thedrow done, please check